cmd/soroban-rpc/internal/db/ledger.go

```diff
@@ -75,6 +75,39 @@ func (r ledgerReader) GetLedger(ctx context.Context, sequence uint32) (xdr.Ledge
 
 // GetLedgerRange pulls the min/max ledger sequence numbers from the meta table.
 func (r ledgerReader) GetLedgerRange(ctx context.Context) (ledgerbucketwindow.LedgerRange, error) {
+	r.db.cache.RLock()
+	latestLedgerSeqCache := r.db.cache.latestLedgerSeq
+	latestLedgerCloseTimeCache := r.db.cache.latestLedgerCloseTime
+	r.db.cache.RUnlock()
+
+	// Make use of the cached latest ledger seq and close time to query only the oldest ledger details.
+	if latestLedgerSeqCache != 0 {
+		query := sq.Select("meta").
+			From(ledgerCloseMetaTableName).
+			Where(
+				fmt.Sprintf("sequence = (SELECT MIN(sequence) FROM %s)", ledgerCloseMetaTableName),
+			)
+		var lcm []xdr.LedgerCloseMeta
+		if err := r.db.Select(ctx, &lcm, query); err != nil {
+			return ledgerbucketwindow.LedgerRange{}, fmt.Errorf("couldn't query ledger range: %w", err)
+		}
+
+		if len(lcm) == 0 {
+			return ledgerbucketwindow.LedgerRange{}, ErrEmptyDB
+		}
+
+		return ledgerbucketwindow.LedgerRange{
+			FirstLedger: ledgerbucketwindow.LedgerInfo{
+				Sequence:  lcm[0].LedgerSequence(),
+				CloseTime: lcm[0].LedgerCloseTime(),
+			},
+			LastLedger: ledgerbucketwindow.LedgerInfo{
+				Sequence:  latestLedgerSeqCache,
+				CloseTime: latestLedgerCloseTimeCache,
+			},
+		}, nil
+	}
+
```

Comment:
Did you measure whether it's worth doing this? It's probably as expensive to query the `MIN` as `MIN+MAX`.

Reply:
In fact, we should be looking at the uses of `GetLedgerRange` to see if all callers are using the oldest seq+closeTime.

Reply:
From what I see `getLatestLedgerSequence` is one of those. It shouldn't be obtaining the oldest seq since it's not using it.

Reply:
@2opremio If you look at the benchmark I poseted here - https://github.com/stellar/soroban-rpc/pull/256#issuecomment-2271592530 - it has reduced the `GetLedgerRange` time by almost half just by querying only the `MIN`. So yes, there is an advantage to using the cache here.

Technically, the `GetLedgerRange` will only be called once in `getLatestLedgerSequence` when the cache is initially empty. On all subsequent requests, it will just return the cached value.